configure.ac

```diff
@@ -5830,6 +5831,12 @@ AC_ARG_WITH(
   [with_readline=readline]
 )
 
+dnl gh-105323: Need to handle the macOS editline as an alias of readline.
+AS_CASE([$ac_sys_system/$ac_sys_release],
+  [Darwin/*], [AC_CHECK_TYPE(Function, [AC_DEFINE([WITH_APPLE_EDITLINE])], [], [#include <readline/readline.h>])],
```

Comment:
@erlend-aasland Is it correct to write the header in this way, IIRC you recommend another way to write header in configure.ac.

Reply:
```suggestion
  [Darwin/*], [AC_CHECK_TYPE([Function],
                             [AC_DEFINE([WITH_APPLE_EDITLINE])],
                             [],
                             [@%:@include <readline/readline.h>])],
```